src/lib.rs

```diff
@@ -23,8 +23,9 @@
 //!
 //! Single-character options are expected to appear on the command line with a
 //! single preceding dash; multiple-character options are expected to be
-//! proceeded by two dashes. Options that expect an argument accept their
-//! argument following either a space or an equals sign. Single-character
+//! proceeded by two dashes. Options that expect a required argument accept such
+//! argument following either a space or an equals sign, but those that expect an
+//! optional argument only accept such argument following an equals sign. Single-character
 //! options don't require the space. Everything after double-dash "--"  argument
```

Comment:
ok this PR isn't good enough, because this won't work `-U=21` it must be either `-U21` or `-U 21`, so even the initial wording that suggested equals worked for both options was flawed for this.

I need to modify this PR to handle the above.